### PR TITLE
Computed attrs for state and status codes

### PR DESCRIPTION
**Pull request by allthesignals**
This PR adds some "computed attributes" for posting into the system. 

kudos to @spencer516 for the mere _7 lines_ of code it took to get a computed attr working.

### CODE REVIEW

File: tests/unit/models/disposition-test.js [hunk 1]
```diff
@@ -1,13 +1,50 @@
...
   // Replace this with your real tests.
-  test('it exists', function(assert) {
+  test('it exists', async function(assert) {
     const store = this.owner.lookup('service:store');
     const model = store.createRecord('disposition', {});
+
     assert.ok(model);
   });
+
+  test('when participant submits/opts out, statuscode is "saved"', async function(assert) {
```
Comment (godfreyyeung) [inline tests/unit/models/disposition-test.js]: Maybe the title should be when participant opts out/submits, statuscode becomes "saved"/"submitted"

File: tests/unit/models/disposition-test.js [hunk 2]
```diff
@@ -1,13 +1,50 @@
...
+    assert.equal(model.statuscode, STATUSCODES.findBy('Label', 'Submitted').Value);
+  });
+
+  test('when statuscode is draft or saved, statecode is "active"', async function(assert) {
+    const store = this.owner.lookup('service:store');
+    const model = store.createRecord('disposition', {
+      statuscode: STATUSCODES.findBy('Label', 'Draft').Value,
+    });
+
+    assert.equal(model.statecode, STATECODES.findBy('Label', 'Inactive').Value);
```
Comment (godfreyyeung) [inline tests/unit/models/disposition-test.js]: Weird, shouldn't statecode be Active if statuscode is Daft? The `statecode()` computed looks right, though...

File: tests/unit/models/disposition-test.js (hunk 1 above)
Comment (allthesignals) [reply tests/unit/models/disposition-test.js]: good catch!

File: tests/unit/models/disposition-test.js (hunk 2 above)
Comment (allthesignals) [reply tests/unit/models/disposition-test.js]: yup!

**Pull request merged**